### DISCUSSION TITLE
offline.php file override

### DISCUSSION THREAD

**Discussion by iorbita** [templaza/astroid-framework#548] [Q&A]
Hi,
is it possible to override **offline.php** file located in **/templates/astroid_template_one/offline.php**?
I tried to copy it into the child template folder (**/media/templates/site/astroid_template_one**) but the file is still not taken into account, any idea why?

Thanks,
Lorenzo

**Comment by Chacapamac** [comment]
@iorbita 
This is what I find.
https://codingwithpops.com/joomla/overriding-the-offline-page

Basically it seem that you can do that.
— You can copy the offline.php file from the /templates/system/ folder to your template folder and modify it as you wish. 
— You also need to change the CSS stylesheet paths in the offline.php file to point to the right place

Tell Us if it work?

It seem that You can use some Joomla plugins ( Search “Offline”), maybe easier?

**Comment by iorbita** [comment]
yes, I found this article too, but the problem is that Astroid framework uses its own offline.php file, so it would be logical that if I copy this file into the child template it should be taken into account, but it isn't...

![astroid-offline_php](https://github.com/templaza/astroid-framework/assets/566221/cf4c371c-bc19-43cd-bbbc-683ff2399999)


Anyway and just for information, I use EasyLayouts from Geraint in all my websites (honestly, I think the best component for layouts overrides…), I asked if it was possible to create layouts to override Joomla core layouts, such as the login page, registration page, [offline page](https://www.easylayouts.net/blog/custom-offline-and-error-pages), ... this should be in place by this evening.

![EL-user-login-override](https://github.com/templaza/astroid-framework/assets/566221/8194c76d-cc9c-4239-9ae8-a39386ea40d3)

The question is for those who don't use the component, and for me to understand how to go about it...

**Reply by Chacapamac** [reply]
**I See, You’re right,** 

Try if a copy of your custom Astroid offline.php place in the media folder to see if it override the template folder one????

**Will be interested to know if it work?????**

**Comment by iorbita** [comment]
> I tried to copy it into the child template folder (/media/templates/site/astroid_template_one) but the file is still not taken into account, any idea why?

...is what I writted in my first post, the file is not taken into account

**Comment by Chacapamac** [comment]
I think you need to do it this way. —> https://codingwithpops.com/joomla/overriding-the-offline-page

**Comment by iorbita** [comment]
the solution you give is for J!3, I'm on J!4... anyway I tried to move the file to other folders that I thought were probably going to take it into account but it doesn't work, ... it would be nice if Sonny could give us a solution :)

**Comment by sonvnn** [comment]
Thank you for feedback. I will check this issue and response you asap.

**Comment by iorbita** [comment]
> Thank you for feedback. I will check this issue and response you asap.

Hi, any news on this? Thanxxxx :)

**Comment by Chacapamac** [comment]
Trying to help...

**offline.php** (in Joomla 4) is found here:
![Screen Shot 2024-02-02 at 8 39 26 AM](https://github.com/templaza/astroid-framework/assets/1907276/5d0d109f-f2ba-47aa-be8c-1dac0992b1f7)

In Joomla 4 —> **offline.css** and **offline_rtl.css** are found here (media folder);
![Screen Shot 2024-02-02 at 8 34 48 AM](https://github.com/templaza/astroid-framework/assets/1907276/7e641b55-737f-4571-9040-513dbf6d6614)

I don’t see specific  **offline.css** and **offline_rtl.css** for Astroid?

• By logic, I think,  you have no choice to replace the _**Root** ➔ **templates** ➔ **yourtemplatename** ➔ **offline.php**_ with your custom one _(and replace it every time the file is overwritten by an Astroid update)_

and add a custom **offline.css** and **offline_rtl.css** to your **css** folder in the **media** folder
_**Root** ➔ **media** ➔ **site** ➔ **yourtemplatename** ➔ **css** ➔ **offline.css/offline_rtl.css**_

**Reply by iorbita** [reply]
...thanks Chacapamac, so for the moment it seems it's not possible to override the offline.php file...
For css files, it's not really a problem because I was going to add css rules in a <style> tag inside the offline.php file.

**Comment by Chacapamac** [comment]
Did you try one of the top review free Joomla extensions to control the “Offline”
https://extensions.joomla.org/category/miscellaneous/offline//

**Comment by iorbita** [comment]
Thanks, I've found a workaround with EasyLayouts, thanks again ;)

**Comment by sonvnn** [comment]
@iorbita I tried to copy offline.php to child-template. It works very well. You can use override offline.php by this way.
![Screen Shot 2024-02-06 at 10 05 25](https://github.com/templaza/astroid-framework/assets/3204103/26373849-2efc-4d6d-86cc-da96bce1b98c)

**Comment by iorbita** [comment]
Hi Sonny,
from your screenshot, I can't see where _"templates"_ folder is located. I assume this folder is in **/media/templates**?

Why do you have two **astroid_template_one** folders? 

Since **astroid_template_one** template is already located in /media/templates, doesn't that automatically make it a child template?

**Comment by Chacapamac** [comment]
> Since astroid_template_one template is already located in /media/templates, doesn't that automatically make it a child template?

No, this how Joomla Templates Systems (should) work now. Any template systems that support Joomla Child templates installtion is split between these 2 folders (root > templates > yourtemplates) and  (root > media > templates > site > yourtemplate)

Meaning that when you originally installed Astroid, the files in both folder are part of the same templates and if you modify any of them you are actually modifying the original template. 

**• Basics Notion on Child Templates**
If you are familiar with the JoomlaRoot > templates > Your_template > HTML folder override system, you know that you can customize many Joomla core layout files and keep the modifications in this folder even if Joomla is updated.

Now, with the “Child Template” ability, you can do the same type of override with ALL files making up a template. It permits you to change anything in a template and keep these changes even if you update the Parent Template. 

When creating a “Child Template” another template folder is created with basically nothing else than a templateDetails.xml file. At this point, the “Child Template” take All of the Parent files and settings to construct its layout.

Any settings changes in the “Child Templates” or added “Parent” corresponding folder/files will override the “Parent”. This way you can override only the settings of the “Child” (Template Manager) or start to override ANYTHING from the “Parent” in the “Child Template” folder. When you update the “Parent” you are also updating the “Child template”, but all your “Child” settings and folder/files override stays untouched.

Now, you have the power to completely modified a Template or simply changed some settings without worrying braking or touching the Parent Template.

**Reply by sonvnn** [reply]
Thank you! This is totally correct!

**Comment by iorbita** [comment]
I know, but my question is about why two template folders (**astroid_template_one** and **astroid_template_one_child**) are in the same media folder. This means that another **astroid_template_one** folder is in _/templates/astroid_template_one_, making 3 folders for the same template and not 2 folders.

**Comment by Chacapamac** [comment]
I just made this graphic overview of installing an Astroid Template, making a child template and the possibility.
Tell me if you see an error?

@sonvnn  You can use it in Astroid documentation if you want. I think it can help user understand the concept rapidly.

![Astroid-files-Install-and-child](https://github.com/templaza/astroid-framework/assets/1907276/e8990c4c-4d94-4806-a6af-9cc80f544614)

**Comment by iorbita** [comment]
I think the best way to understand how Joomla and Astroid work together to build a child template is to install a blank version of Joomla/Astroid and play around with it (that's what I did). Chacapamac diagram is a helpful resource for understanding folder hierarchies and how they work.

However, I think this procedure is out of scope for the most ordinary Joomla user and you need to find a real reason to use a child template, but on the other hand will be a godsend for anyone who wants to customize framework and template in a safe and professional way.

**Comment by rinka88** [comment]
@sonvnn And how do I override this file within the template itself to modify certain blocks?
https://github.com/templaza/astroid-framework/blob/master/framework/layouts/system/offline.php
so that it doesn't revert during the next update

**Reply by rinka88** [reply]
I wouldn't want to display the login form on offline pages.
